### PR TITLE
231031 medlink

### PR DESCRIPTION
**Pull request by ycq091044**
merged from #228

### CODE REVIEW

File: pyhealth/models/base_model.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 
 # TODO: add support for regression
-VALID_MODE = ["binary", "multiclass", "multilabel"]
+VALID_MODE = [None, "binary", "multiclass", "multilabel"]
```
Comment (ycq091044) [inline pyhealth/models/base_model.py]: Could we replace "None" with "Ranking" or something else?

File: pyhealth/metrics/ranking.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from typing import List, Dict
+
+
+def ranking_metrics_fn(qrels: Dict[str, Dict[str, int]],
```
Comment (ycq091044) [inline pyhealth/metrics/ranking.py:4]: Could you add an example for this ranking function? How to use it, what is query, result, and kvalues?

File: pyhealth/trainer.py [hunk 3]
```diff
@@ -304,12 +306,22 @@ def evaluate(self, dataloader) -> Dict[str, float]:
         Returns:
             scores: a dictionary of scores.
         """
-        y_true_all, y_prob_all, loss_mean = self.inference(dataloader)
-
-        mode = self.model.mode
-        metrics_fn = get_metrics_fn(mode)
-        scores = metrics_fn(y_true_all, y_prob_all, metrics=self.metrics)
-        scores["loss"] = loss_mean
+        if self.model.mode is not None:
```
Comment (ycq091044) [inline pyhealth/trainer.py:309]: change into your new mode, but do not use  "None"

**Review by ycq091044 — COMMENTED** [review]
LGTM, please address the minor comments and add docstring if possible.

File: pyhealth/models/base_model.py (hunk 1 above)
Comment (zzachw) [reply pyhealth/models/base_model.py]: The evaluation pipeline for ranking is not compatible with Pyhealth's Trainer. That's why I add a None option here. I will add more instructions.

**Comment by zzachw** [comment]
Resolved. Can you take a look? @ycq091044

**Pull request merged**